src/ui/src/renderer_widget.cpp

```diff
@@ -82,3 +94,52 @@ QSurfaceFormat surface_format(const QSurfaceFormat::FormatOptions options) {
 	surface_format.setStencilBufferSize(8);
 	return surface_format;
 }
+
+void RendererWidget::mousePressEvent(QMouseEvent* event) { 
+	previous_mouse_pos_ = event->pos();
+	event->accept();
+}
+
+
+void RendererWidget::mouseReleaseEvent(QMouseEvent* event) { 
+	event->accept(); 
+}
+
+void RendererWidget::mouseMoveEvent(QMouseEvent* event) {
+	auto clicked_buttons = event->buttons();
+	// if button still pressed we use it and don't mention other clicked buttons
+	if (current_pressed_button_ & clicked_buttons)
+		clicked_buttons = current_pressed_button_;
+	bool action_requiered = true;
```

Comment:
Review this function please. `action_required` (don't forget about typo) have strange usage